api/handlers_datasets.go

```diff
@@ -234,17 +234,141 @@ func (s *server) DatasetShowHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write(j)
 }
 
+// DatasetPromoteHandler promotes a dataset
+// If this is an original dataset, it will be finalized (if not already finalized)
+// If this is a derivative, it will be promoted to an original and instantly finalized
+func (s *server) DatasetPromoteHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := vars["account"]
+	id := vars["id"]
+
+	service, ok := s.datasetServices[account]
+	if !ok {
+		msg := fmt.Sprintf("account not found: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+
+	input := struct {
+		ModifiedBy string `json:"modified_by"`
```

Comment:
I wonder if this should actually be that header that we use in the server-api... it seems better to standardize on "who" is doing something for all of our APIs.

Reply:
You mean `X-Forwarded-User`? Good point, it seemed a little weird to just pass this one request parameter. We could also use that for the `modified_by` user for all _update_ calls. I think we're already injecting that header automatically in all requests from the UI.